vllm/executor/msgspec_utils.py

```diff
@@ -0,0 +1,36 @@
+from array import array
+from typing import Any, Type
+
+from vllm.sequence import SequenceData
+
+
+def encode_hook(obj: Any) -> Any:
+    """Custom msgspec enc hook that supports array types.
```

Comment:
Question: why do we treat array differently?

Reply:
by default, msgspec cannot handle array serialization. it is a simple wrapper on c array

Reply:
Add that to the docstring?